frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.unit.spec.tsx

```diff
@@ -1,43 +1,67 @@
 import { checkNotNull } from "metabase/core/utils/types";
 import { getMetadata } from "metabase/selectors/metadata";
-import { SchemaId, TableId } from "metabase-types/api";
+import { InitialSyncStatus, Schema, Table } from "metabase-types/api";
 import { createMockDatabase, createMockTable } from "metabase-types/api/mocks";
 import { createMockState } from "metabase-types/store/mocks";
 import { createMockEntitiesState } from "__support__/store";
-import { renderWithProviders, screen } from "__support__/ui";
+import { renderWithProviders, screen, within } from "__support__/ui";
 import DataSelectorTablePicker from "./DataSelectorTablePicker";
 
 const TEST_TABLE = createMockTable();
 const TEST_DATABASE = createMockDatabase({ tables: [TEST_TABLE] });
+const NOT_SYNCED_DB_STATUSES: InitialSyncStatus[] = ["aborted", "incomplete"];
 
 interface SetupOpts {
-  schemaIds?: SchemaId[];
-  tableIds?: TableId[];
+  schemas?: Schema[];
+  tables?: Table[];
 }
 
-const setup = ({ schemaIds = [], tableIds = [] }: SetupOpts = {}) => {
+const setup = ({ schemas = [], tables = [] }: SetupOpts = {}) => {
+  const TEST_DATABASE = createMockDatabase({ tables: tables });
   const state = createMockState({
     entities: createMockEntitiesState({
       databases: [TEST_DATABASE],
     }),
   });
   const metadata = getMetadata(state);
   const database = checkNotNull(metadata.database(TEST_DATABASE.id));
-  const schemas = schemaIds.map(id => checkNotNull(metadata.schema(id)));
-  const tables = tableIds.map(id => checkNotNull(metadata.table(id)));
+  const metadataSchemas = schemas.map(schema =>
+    checkNotNull(metadata.schema(schema.id)),
+  );
+  const metadataTables = tables.map(table =>
+    checkNotNull(metadata.table(table.id)),
+  );
 
   renderWithProviders(
     <DataSelectorTablePicker
       selectedDatabase={database}
-      schemas={schemas}
-      tables={tables}
+      schemas={metadataSchemas}
+      tables={metadataTables}
       onChangeTable={jest.fn()}
     />,
     { storeInitialState: state },
   );
 };
 
 describe("DataSelectorTablePicker", () => {
+  it.each(NOT_SYNCED_DB_STATUSES)(
+    "render a loading spinner when a table has initial_sync_status='incomplete'",
```

Comment:
nit: the test name is probably written before it was refactored to `it.each`. You can see what special symbol we can use to reference the values inside `it.each` [in the document](https://jestjs.io/docs/api#1-testeachtablename-fn-timeout).

```suggestion
    "render a loading spinner when a table has initial_sync_status='%s'",
```

Reply:
Fixed